### ISSUE TITLE
Support providing schema via json string or JsonObject

### ISSUE THREAD

**Issue by snowbldr** [bkbnio/kompendium#303]
**Describe the solution you'd like**
I am using jakarta validation annotations on my data classes. Using these allows me to perform validation and generate json schema using victools json-schema-generator with the jakarta module. This all works very well and is pretty straight forward to use, even in kotlin. 

I would like to use these generated schemas to configure kompendium, instead of passing a KType, as there is currently no support for specifying max/min/pattern etc. Even if there were support, I would prefer to use the jakarta annotations for the validation that I get with it. 

**Describe alternatives you've considered**
As far as I can tell, the only alternative would be manually building the openapi schema and setting everything up by hand.

**Additional context**
I've started working on this a little bit, but I'm unsure of a great way to proceed. My initial thought was to change the MediaType to accept a JsonObject, then in the helpers convert the type to a JsonSchema, then to a JsonObject. This would allow the user to set either a type or a JsonObject directly. The one thing I'm not totally sure about is how to handle the ref situation, and to ensure that the same schemas aren't re-used.

I'd like input on whether that's a good idea or not, and if you have any other ideas about a good way to support this feature.

**Comment by unredundant** [comment]
Hmm... bit hard to follow the proposed solution without some code. 

I have a solution that I _think_ will work, but I haven't tried it so no promises :) 

If you generate all of these types, and have endpoints for each schema definition (ie `/schemas/my_cool_type.json`) I believe you could then set up a Kompendium custom type (see [here](https://github.com/bkbnio/kompendium/blob/main/playground/src/main/kotlin/io/bkbn/kompendium/playground/CustomTypePlayground.kt#L50) for example) you could set up a `ReferenceDefinition` to point to where you are hosting the schema definition.  Kompendium doesn't perform introspection on types that have already been stored in it's cache (to support custom types and recursion), so you should be able to then just use Kompendium as normal, and it will just never perform the type introspection.

Hope that all made sense :) 

Also happy to take a look at any proposed solutions if you can get a simple example working!

**Comment by snowbldr** [comment]
Sounds like it could work. I'll try out using the custom types and see how that goes, and I'll post some code as soon as I get it working

**Comment by snowbldr** [comment]
BTW, I love these test classes https://github.com/bkbnio/kompendium/blob/main/core/src/testFixtures/kotlin/io/bkbn/kompendium/core/fixtures/TestModels.kt#L77

😆

**Comment by unredundant** [comment]
All praise the Gibbits 🙌

**Comment by unredundant** [comment]
hey just follow up here, were you able to get this sorted out?

**Comment by snowbldr** [comment]
I have not sorted it yet. Hopefully I will have some time in the next week or two to get back to this. I'll post an update here once I make some progress

**Comment by unredundant** [comment]
I'll still welcome any PRs on this, but going to close this issue as wont fix due to inactivity
